apiserver/apiserver.go

```diff
@@ -880,6 +903,14 @@ func (srv *Server) trackRequests(handler http.Handler) http.Handler {
 	})
 }
 
+func (srv *Server) healthHandler(w http.ResponseWriter, req *http.Request) {
+	srv.mu.Lock()
+	status := srv.healthStatus
+	srv.mu.Unlock()
+
+	fmt.Fprintf(w, "%s\n", status)
```

Comment:
doesn't this mean we return 200 under every case? Shouldn't we return a different HTTP response if the server isn't ready?